Index: alsa-lib-1.2.2/src/conf/cards/USB-Audio.conf
===================================================================
--- alsa-lib-1.2.2.orig/src/conf/cards/USB-Audio.conf
+++ alsa-lib-1.2.2/src/conf/cards/USB-Audio.conf
@@ -43,6 +43,7 @@ USB-Audio.pcm.iec958_device {
 	# The below don't have digital in/out, so prevent them from being opened.
 	"Andrea PureAudio USB-SA Headset" 999
 	"Blue Snowball" 999
+	"Cmedia Audio" 999
 	"HP Digital Stereo Headset" 999
 	"GN 9330" 999
 	"Logitech Speaker Lapdesk N700" 999
